### PR TITLE
refactor(docs)!: start working on sphinx differentiation

### LINKED ISSUE #28: Split documentation engine sphinx into  sphinx-md and sphinx-rst
ideally if sphinx is selected, it should show an option for source format md or rst

### PR DESCRIPTION
**Pull request by luabida**
## Pull Request description

add sphinx .md and .rst options.

I've changed the docs post_gen system, thats why its a braking change

fix #28 

## How to test these changes

...

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [x] new feature
- [x] maintenance

About this PR:
- [x] it includes tests.
- [x] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [x] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [x] New and old tests passed locally.

## Additional information

still missing changes on Makefile, creating a draft PR

BREAKING CHANGE: new docs post hook system

### CODE REVIEW

**Comment by luabida** [comment]
As pointed the worflow on CI, the smoke tests also need refactor

**Comment by luabida** [comment]
@xmnlab for some reason, the method `shutil.move()` is moving the `conf.py` file as `conf.py*`, did you ever see this behavior before?
image

**Comment by xmnlab** [comment]
> As pointed the workflow on CI, the smoke tests also need refactor

could explain that a bit more pls?

**Comment by luabida** [comment]
@xmnlab they were refactored. It's passing now

**Comment by xmnlab** [comment]
ok .. as there are a lot of files .. it would take me some time to review.. I hope to finish the review by tomorrow

**Comment by xmnlab** [comment]
https://github.com/cookiecutter/cookiecutter/issues/1438#issuecomment-1359487753

**Comment by xmnlab** [comment]
https://github.com/cookiecutter/cookiecutter/pull/1645

**Comment by xmnlab** [comment]
@luabida could you try to use this approach? https://github.com/cookiecutter/cookiecutter/issues/1438#issuecomment-1359487753

so we can have just "sphinx" "mkdocs" and "jupyter-book" and if the sphinx option is selected we can show an option called "format" and the values "markdown" or "reStructuredText"

**Comment by luabida** [comment]
@xmnlab sure, I think the new structure in this PR can be reused as well. I'll take a look into it

File: pyproject.toml [hunk 1]
```diff
@@ -48,14 +48,16 @@ isort = "^5"
...
 mypy = "^0.931"
-cookiecutter = "^1.7.3"
 pytest-cov = "^3.0.0"
 pytest-cookies = "^0.6.1"
 Sphinx = "^4.4.0"
 Flake8-pyproject = "^0.9.0"
 jupyter-book = "^0.12.3"
 myst-parser = "^0.15"
 
+[tool.poetry.group.dev.dependencies]
```
Comment (luabida) [inline pyproject.toml:58]: @xmnlab seems like poetry now has group dependencies, which I think `--dev` is just one of the possible groups, as you can see a "test" group in this example: https://python-poetry.org/docs/master/managing-dependencies/. But I didn't find any reference to poetry version in the project, so I assume it is correct..

Comment (xmnlab) [reply pyproject.toml:58]: Yep I saw that some days ago, but never tried. But for us just dev should be enough

**Comment by xmnlab** [comment]
@luabida when you want to move this PR forward, it uses already cookieninja, maybe it would have to have conditional options.

**Comment by xmnlab** [comment]
thanks for working on this PR @luabida 
we probably will move this change in a bit different way
I am closing this PR for now

**Pull request closed without merging**